### PR TITLE
Module 2

### PR DESCRIPTION
**Pull request by malicuss**
Module 2

### CODE REVIEW

**Comment by Ungerfall** [comment]
Checklist:
* [ ] Ctor injection
* [ ] Tests project
* [ ] Tests
* [ ] Category image action
* [ ] Links to images on the categories page
* [ ] Upload image
* [ ] Routing
* [ ] Custom middleware for caching
  * [ ] Cache on disk
  * [ ] Configuration support
* [ ] Logging filter

File: SimpleWebApp/Services/ActionLoggingFilter.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    public ActionLoggingFilter(ILogger<ActionLoggingFilter> logger,
+        IOptions<AppOptions> options)
+    {
+        _logger = logger;
+        _logParameters = options.Value.LogActionParameters;
+    }
+    public void OnActionExecuting(ActionExecutingContext context)
+    { 
+        _logger.LogCritical(String.Format("{0} - invoked", context.ActionDescriptor.DisplayName));
```
Comment (Ungerfall) [inline SimpleWebApp/Services/ActionLoggingFilter.cs]: Severity level should be lower

File: SimpleWebApp/Middleware/ImageCache.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using SimpleWebApp.Helpers;
+
+namespace SimpleWebApp.Middleware;
+
+public class ImageCache
+{
+    private readonly ILogger<ImageCache> _logger;
+    private readonly RequestDelegate _next;
+    private readonly ICacher _cacher;
```
Comment (Ungerfall) [inline SimpleWebApp/Middleware/ImageCache.cs:15]: Cache is not utilized

**Review by Ungerfall — COMMENTED** [review]
Checklist:
* [x] Ctor injection
* [x] Tests project
* [x] Tests
* [x] Category image action
* [x] Links to images on the categories page
* [x] Upload image
* [x] Routing
* [x] Custom middleware for caching
  * [x] Cache on disk
  * [x] Configuration support
* [x] Logging filter

File: SimpleWebApp/Middleware/ImageCache.cs (hunk 2 above)
Comment (malicuss) [reply SimpleWebApp/Middleware/ImageCache.cs:9]: fixed

Comment (malicuss) [reply SimpleWebApp/Middleware/ImageCache.cs:15]: branch updated with fix

**Review by Ungerfall — APPROVED** [review]

**Pull request merged**